vyxal/LazyList.py

```diff
@@ -269,3 +269,10 @@ def reversed(self):
         self.generated += list(itertools.tee(self.raw_object)[-1])
         for item in self.generated[::-1]:
             yield item
+
+    def is_empty(self):
+        try:
+            next(self)
```

Comment:
This throws away the result of next

Reply:
Try has_ind(0) maybe

Reply:
```py
 def has_ind(self, ind: int):
    """Whether or not this list is long enough for index `ind`"""
    if ind < len(self.generated):
        return 0 <= ind
    else:
        for _ in range(ind - len(self.generated) + 1):
            try:
                next(self)
            except StopIteration:
                return False
        return True
```

by that logic, `has_ind` should also throw away the result of `next`.

But it doesn't, because `next` is smart enough to store the item in `self.generated`

Reply:
Oh right